bigquery_etl/backfill/shredder_mitigation.py

```diff
@@ -50,30 +67,187 @@ class DataTypeGroup(Enum):
+
+    @data_type.validator
+    def validate_data_type(self, attribute, value):
+        """Check that the type of data_type is as expected."""
+        if not isinstance(value, DataTypeGroup):
+            raise ValueError(f"Invalid {value} with type: {type(value)}.")
+
+    @column_type.validator
+    def validate_column_type(self, attribute, value):
+        """Check that the type of parameter column_type is as expected."""
+        if not isinstance(value, ColumnType):
+            raise ValueError(f"Invalid data type for: {value}.")
+
+    @status.validator
+    def validate_status(self, attribute, value):
+        """Check that the type of parameter column_status is as expected."""
+        if not isinstance(value, ColumnStatus):
+            raise ValueError(f"Invalid data type for: {value}.")
+
+
+@attr.define
+class Subset:
+    client: bigquery.Client()
+    destination_table: str = attr.field()
+    suffix: str = attr.field(converter=attr.converters.default_if_none(""))
+    query_cte: str = attr.field(converter=attr.converters.default_if_none(""))
+    dataset: str = attr.field(default=TEMP_DATASET)
+    project_id: str = attr.field(default=DEFAULT_PROJECT_ID)
+    expiration_days: Optional[float] = attr.field(default=None)
+
+    @destination_table.validator
+    def validate_table_version(self, attribute, value):
+        """Raise an exception if the version table name doesn't end with an integer."""
+        match = re.search(r"v(\d+)$", self.destination_table)
+        try:
+            version = int(match.group(1))
+            if not isinstance(version, int) or version <= 0:
+                raise click.ClickException(
+                    f"Destination table {value} must end with a positive version number."
+                )
+        except AttributeError:
+            f"Invalid table version for {value}"
+
+    @property
+    def expiration_ms(self) -> Optional[float]:
+        """Convert partition expiration from days to milliseconds."""
+        if self.expiration_days is None:
+            return None
+        return int(self.expiration_days * 86400000)
+
+    @property
+    def name(self):
+        if self.suffix is not None and self.suffix != "":
+            return f"{self.destination_table}{self.suffix}"
+        else:
+            return self.destination_table
+
+    @property
+    def version(self):
+        match = re.search(r"v(\d+)$", self.destination_table)
+        return int(match.group(1))
+
+    @property
+    def name_previous_version(self):
+        """Return the table name for the previous version of the table."""
+        return f"{self.destination_table[:-1]}{int(self.version)-1}"
+
+    @property
+    def full_table_id(self):
+        return f"{self.project_id}.{self.dataset}.{self.name}"
+
+    @property
+    def partitioning(self) -> str:
+        metadata = Metadata.from_file(
+            Path("sql") / self.project_id / self.dataset / self.name / METADATA_FILE
```

Comment:
sure this would be `self.name` and not `self.destination_table`? because whatever the suffix might be .. it's probably not in the file path???

Reply:
Indeed! The suffix is generating confusion and it's only available in case it's required to save the subsets in temp tables. Since it's not the case the suffix is removed and this now uses `destination_table`.